query-engine/query-engine-wasm/build.sh

```diff
@@ -30,114 +42,85 @@ if [[ -z "${WASM_BUILD_PROFILE:-}" ]]; then
     fi
 fi
 
-echo "Using build profile: \"${WASM_BUILD_PROFILE}\"" 
-
-if ! command -v wasm-pack &> /dev/null
-then
-    echo "wasm-pack could not be found, installing now..."
-    curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
+if [ "$WASM_BUILD_PROFILE" = "dev" ]; then
+    WASM_TARGET_SUBDIR="debug"
+else
+    WASM_TARGET_SUBDIR="release"
 fi
 
+echo "Using build profile: \"${WASM_BUILD_PROFILE}\"" 
+
 echo "ℹ️  Configuring rust toolchain to use nightly and rust-src component"
 rustup default nightly-2024-01-25 
 rustup target add wasm32-unknown-unknown
 rustup component add rust-src --target wasm32-unknown-unknown
-
-
-echo "Building query-engine-wasm using $WASM_BUILD_PROFILE profile"
 export RUSTFLAGS="-Zlocation-detail=none"
-CARGO_PROFILE_RELEASE_OPT_LEVEL="z" wasm-pack build "--$WASM_BUILD_PROFILE" --target "$OUT_TARGET" --out-dir "$OUT_FOLDER" --out-name query_engine . \
-    -Zbuild-std=std,panic_abort -Zbuild-std-features=panic_immediate_abort
+CARGO_TARGET_DIR=$(cargo metadata --format-version 1 | jq -r .target_directory)
+
+build() {
+    local CONNECTOR="$1"
+    echo "🔨 Building $CONNECTOR"
+    CARGO_PROFILE_RELEASE_OPT_LEVEL="z" cargo build \
+        -p query-engine-wasm \
+        --profile "$WASM_BUILD_PROFILE" \
+        --features "$CONNECTOR" \
+        --target wasm32-unknown-unknown \
+        -Zbuild-std=std,panic_abort -Zbuild-std-features=panic_immediate_abort
+
+    local IN_FILE="$CARGO_TARGET_DIR/wasm32-unknown-unknown/$WASM_TARGET_SUBDIR/query_engine_wasm.wasm"
+    local OUT_FILE="$OUT_FOLDER/$CONNECTOR/query_engine_bg.wasm"
+
+    wasm-bindgen --target "$OUT_TARGET" --out-name query_engine --out-dir "$OUT_FOLDER/$CONNECTOR" "$IN_FILE"
+    optimize "$OUT_FILE"
+
+    if ! command -v wasm2wat &> /dev/null; then
+        echo "Skipping wasm2wat, as it is not installed."
+    else
+        wasm2wat "$OUT_FILE" -o "./query_engine.$CONNECTOR.wat"
+    fi
+}
```

Comment:
There are two different ways in which engines are built: On the one hand nix is outputing the `IN_FILE`, using the provider as the file name and putting all engines for the different providers in the same directory.  On the other hand, this script reads those files and outputs the final version of the engine in a different directory per provider.

I guess you thought of making the nix output align better with the way engines are finally built by the `build.sh` script. Doing so will make easier to reason about the nix file, as for instance you could correlate in nix how the engines are built, based for instance on how an import of the engine is imported in a script such as the executor `wasm.ts` module, but I bet there is something impractical about it. If so, what's that?

Reply:
>There are two different ways in which engines are built: On the one hand nix is outputing the IN_FILE, using the provider as the file name and putting all engines for the different providers in the same directory

That's not quite true. Engine in nix is built by calling the `build.sh` script and published using the same layout we have in `pkg`. That script that puts them all into same directory is a different nix package (wasm-engine-gz) and it is used only for measuring engine size over time in [this chart](https://prisma.github.io/prisma-engines/engines-size/).

Reply:
Oh, yes, you are right. Thanks for clarifying.